src/server/game/Conditions/ConditionMgr.cpp

```diff
@@ -284,7 +284,11 @@ bool Condition::Meets(ConditionSourceInfo& sourceInfo) const
         }
         case CONDITION_NEAR_CREATURE:
         {
-            condMeets = GetClosestCreatureWithEntry(object, ConditionValue1, (float)ConditionValue2, bool(!ConditionValue3)) ? true : false;
+            Creature* crea = GetClosestCreatureWithEntry(object, ConditionValue1, (float)ConditionValue2, bool(!ConditionValue3));
+            if (!crea || (ConditionValue3 && crea->getDeathState() == DEAD))
```

Comment:
3rd argument controls this - but here you must remember what deathstate values mean
CORPSE = is dead but still visible/spawned
DEAD = despawned, waiting for respawn

If i'm reading this right, this check is for the case when GetClosestCreatureWithEntry is allowed to return dead creatures via ConditionValue3